Assets/ComethSDK/Scripts/Services/EoaFallbackService.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Text;
+using System.Threading.Tasks;
+using ComethSDK.Scripts.HTTP;
+using ComethSDK.Scripts.Tools;
+using ComethSDK.Scripts.Tools.Signers;
+using ComethSDK.Scripts.Types;
+using Nethereum.Signer;
+using UnityEngine;
+
+namespace ComethSDK.Scripts.Services
+{
+	public static class EoaFallbackService
+	{
+		public static async Task<EncryptionData> EncryptEoaFallback(string walletAddress,
+			string privateKey, string salt)
+		{
+			var encodedWalletAddress = Encoding.UTF8.GetBytes(walletAddress);
+			var encodedSalt = Encoding.UTF8.GetBytes(salt);
+
+			var encryptionKey =
+				await CryptoService.Pbkdf2(encodedWalletAddress, encodedSalt, Constants.PBKDF2_ITERATIONS);
+
+			var encodedPrivateKey = Encoding.UTF8.GetBytes(privateKey);
+
+			var iv = CryptoService.GetRandomIV();
+
+			var encryptedPrivateKey = await CryptoService.EncryptAESCBC(encryptionKey, iv, encodedPrivateKey);
+
+			return new EncryptionData
+			{
+				encryptedPrivateKey = Convert.ToBase64String(encryptedPrivateKey),
+				iv = Convert.ToBase64String(iv)
+			};
+		}
+
+		public static async Task<string> DecryptEoaFallback(
+			string walletAddress, byte[] encryptedPrivateKey, byte[] iv, string salt)
+		{
+			var encodedWalletAddress = Encoding.UTF8.GetBytes(walletAddress);
+			var encodedSalt = Encoding.UTF8.GetBytes(salt);
+
+			var encryptionKey =
+				await CryptoService.Pbkdf2(encodedWalletAddress, encodedSalt, Constants.PBKDF2_ITERATIONS);
+
+			var privateKey = await CryptoService.DecryptAESCBC(encryptionKey, iv, encryptedPrivateKey);
+
+			return Encoding.UTF8.GetString(privateKey);
+		}
+
+		public static async Task<Signer> GetSigner(API api, string provider, string walletAddress,
+			string encryptionSalt)
+		{
+			var storagePrivateKey = await GetSignerLocalStorage(walletAddress, encryptionSalt);
+
+			if (string.IsNullOrEmpty(storagePrivateKey))
+				throw new Exception("New Domain detected. You need to add that domain as signer.");
+
+			var storageSigner = new Signer(new EthECKey(storagePrivateKey));
+
+			var isOwner = await SafeService.IsSigner(storageSigner.GetAddress(), walletAddress, provider, api);
+
+			if (!isOwner) throw new Exception("New Domain detected. You need to add that domain as signer.");
+
+			return storageSigner;
+		}
+
+		public static async Task<string> GetSignerLocalStorage(string walletAddress, string encryptionSalt)
+		{
+			if (string.IsNullOrEmpty(encryptionSalt)) encryptionSalt = Constants.DEFAULT_ENCRYPTION_SALT;
+
+			var localStorageV1 = PlayerPrefs.GetString($"cometh-connect-{walletAddress}");
+
+			if (!string.IsNullOrEmpty(localStorageV1))
```

Comment:
I would have created a dedicated method for this logic and called it independently of the GetSigner method. 

Migrating should not be a feature of the GetSigner method. 

The migration method could be called in the Awake method.

Reply:
Yes it would be better, i will change that and do the migration check before the get signer in the Connect